src/coord/src/sink_connector.rs

```diff
@@ -107,62 +117,65 @@ fn get_latest_ts(
         )
     })?;
 
-    let mut latest_message = None;
-    while let Some(message) = get_next_message(consumer, timeout)? {
-        latest_message = Some(message);
+    let (lo, hi) = consumer
+        .fetch_watermarks(consistency_topic, 0, timeout)
+        .map_err(|e| {
+            anyhow!(
+                "Failed to fetch metadata while reading from consistency topic: {}",
+                e
+            )
+        })?;
+
+    // Empty topic
+    if hi == 0 {
```

Comment:
I can't tell if we still need this early exit now that we have the eof thing - if we do, maybe a longer comment as to why would be helpful?